src/lib.rs

```diff
@@ -39,9 +42,24 @@ where
     }
 }
 
+#[derive(Debug)]
+struct SerialTransport<TX>(TX);
+
+impl<TX, E> MidiTransport for SerialTransport<TX>
+where
+    TX: serial::Write<u8, Error = E>,
+    E: Debug,
+{
+    type Error = E;
+
+    fn write(&mut self, value: u8) -> Result<(), Self::Error> {
+        block!(self.0.write(value))
+    }
+}
+
```

Comment:
maybe just a preference but, i don't believe you need this struct at all:

```
impl<TX, E> MidiTransport for TX
where
    TX: serial::Write<u8, Error = E>,
    E: Debug,
{
    type Error = E;

    fn write(&mut self, value: u8) -> Result<(), Self::Error> {
        block!(self.write(value))
    }
}
```

Reply:
That would be my preference too, and I tried this first. But you can only implement traits when the trait or the type that implements it is in your own crate. In this case serial::Write is from the embedded-hal crate and MidiTransport is from the midi-convert crate. So rust didnt let me do that :-( .